Game Development Done! Kindly review

ayoghgugyuy opened the pull request:
kindly leave reviews here please

LouieUni commented:
Overall a great game that runs as intended - 
 Positives:
    Good Object-Oriented Design:
        The code follows OOP principles well, with separate classes for Game, Player, and Room.
        Methods and constructors are logically organized.
    Basic Error Handling is Present:
        The try-catch block in Program.cs ensures that unexpected errors are caught and displayed.
        The TakeDamage method prevents negative damage, which is a nice touch.
    Engaging Gameplay Elements:
        The game includes traps, inventory management, and movement between rooms, making it more interactive.
        The UseItem method allows for healing and key-based interactions.

Areas for improvement -
 Although there arent many adding comments would help increase readibility and to understand what does what, there are a few but adding some extra in game.cs and room.cs would help greatly.

currently in Game.cs, user input (such as go <direction> or use <item>) is not strictly validated before processing.
If a player types an invalid command (like take sword sword, or "go right" when only saying north works), it could cause unexpected behavior. a fix for this would be after the user says the wrong command, give them a list of commands that would work.

Final thoughts - Overall a very good project that meets all the criteria even with some extra stuff, but with a couple tweaks could helps make it even better.

ayoghgugyuy commented:
> **Reviewer - Ashhar Iqbal (29458316)**
> 
> **Overview:** • Compiles and runs – Everything looks properly structured with no obvious syntax errors. • Player can explore multiple rooms – The game allows movement between different rooms. • Rooms have different interactions – Some rooms contain items, some have traps. • Room descriptions and inventory items – The GetDescription method properly describes items and exits. • Handles incorrect commands – The HandleInput method provides feedback on invalid commands. • Uses OOP (Classes, Methods, Encapsulation) – Good use of classes (Game, Player, Room), encapsulation, and methods.
> 
> **Improvements:** • Improve user experience by adding Room Names and not only Room Descriptions. This will allow the player to know where they are • Your game has a trap system, so you could add a trap warning system which notifies the player before stepping into a trap e.g.
> 
> Console.WriteLine($"{player.Name} hears a faint clicking sound... A TRAP!"); player.TakeDamage(TrapDamage);
> 
> **Final thoughts:** • Overall it is quite a good project, with the criteria all met. I like the trap system you have also implemented, and with some small tweaks to your code you can improve the overall user experience!

Changes made as per feedback: Added Room Name display for better user navigation and included trap warning messages before applying damage.

ayoghgugyuy commented:
> Overall a great game that runs as intended - Positives: Good Object-Oriented Design: The code follows OOP principles well, with separate classes for Game, Player, and Room. Methods and constructors are logically organized. Basic Error Handling is Present: The try-catch block in Program.cs ensures that unexpected errors are caught and displayed. The TakeDamage method prevents negative damage, which is a nice touch. Engaging Gameplay Elements: The game includes traps, inventory management, and movement between rooms, making it more interactive. The UseItem method allows for healing and key-based interactions.
> 
> Areas for improvement - Although there arent many adding comments would help increase readibility and to understand what does what, there are a few but adding some extra in game.cs and room.cs would help greatly.
> 
> currently in Game.cs, user input (such as go or use ) is not strictly validated before processing. If a player types an invalid command (like take sword sword, or "go right" when only saying north works), it could cause unexpected behavior. a fix for this would be after the user says the wrong command, give them a list of commands that would work.
> 
> Final thoughts - Overall a very good project that meets all the criteria even with some extra stuff, but with a couple tweaks could helps make it even better.

Thanks a lot for your suggestions! I've added helpful comments in both Room.cs and Game.cs to make the code easier to understand. Also, I improved how the game handles wrong commands—now it shows clear messages and suggests what the player can do instead. Let me know if there's anything else you'd like me to work on!

ayoghgugyuy commented:
Implemented all the changes suggested.

ayoghgugyuy commented:
Implemented all the changes suggested above.
